src/openrct2/peep/Staff.cpp

```diff
@@ -727,10 +727,14 @@ static uint8_t staff_direction_surface(Peep* peep, uint8_t initialDirection)
 
         direction &= 3;
 
-        if (fence_in_the_way(peep->next_x, peep->next_y, peep->next_z, peep->next_z + 4, direction))
+        if (fence_in_the_way(
+                { peep->next_x, peep->next_y, peep->next_z * COORDS_Z_STEP, (peep->next_z * COORDS_Z_STEP) + PATH_HEIGHT },
```

Comment:
This isn't the path height it is the peep clearance height.

Reply:
Add this constant to `Peep.h`

Reply:
Yes